core/prisma/migrations/20241017223037_add_new_membership_tables/migration.sql

```diff
@@ -0,0 +1,123 @@
+/*
+  Warnings:
+
+  - You are about to drop the column `role` on the `member_groups` table. All the data in the column will be lost.
+
+*/
+-- AlterTable
+ALTER TABLE "member_groups" DROP COLUMN "role";
+
+-- CreateTable
+CREATE TABLE "community_memberships" (
+    "id" TEXT NOT NULL DEFAULT gen_random_uuid(),
+    "role" "MemberRole" NOT NULL,
+    "communityId" TEXT NOT NULL,
+    "userId" TEXT,
+    "memberGroupId" TEXT,
+    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updatedAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+    CHECK ( num_nonnulls("userId", "memberGroupId") = 1 )
```

Comment:
We can't express this constraint in prisma, but it should probably be documented somewhere more discoverable than this specific migration